lib/awesome_spawn.rb

```diff
@@ -111,6 +110,13 @@ def build_command_line(command, params = nil)
 
   private
 
+  # extract keys (even those in an array)
+  # useful for file descriptors
+  def flat_keys(options)
+    keys = options.keys.partition { |key| key.respond_to?(:each) }
+    keys.first.empty? ? keys.last : (keys.first.flatten + keys.last)
```

Comment:
This reads really funny to me because typically when you use partition, you put it into two separate variables.  Having `keys` makes it harder to understand...I would expect this to read something like the following, which to me is clearer.

```ruby
arrays, non_arrays = options.keys.partition { |key| key.respond_to?(:each) }
arrays.empty? ? non_arrays : arrays.flatten + non_arrays
```

Even so, it seems really complex...can't you just do...

```ruby
options.keys.flatten
```